subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/cell.go

```diff
@@ -40,14 +41,34 @@ type Cell interface {
 	SetParent(Cell)
 	GetChildren() CellList
 	SetChildren(CellList)
+	AtOrHigherThanNode() bool
+
+	IncreaseFreeGpuNum(int32)
+	GetFreeGpuNumForPriority(CellPriority) int32
+	GetUsedGpuNumAtPriority(CellPriority) int32
+	IncreaseUsedGpuNumAtPriority(CellPriority, int32)
+	GetUsedGpuNumSamePriority() int32
+	GetUsedGpuNumOtherPriority() int32
+	SetUsedGpuNumForPriority(CellPriority)
+	SetUsedGpuNumAllPriority(CellPriority)
 }
 
 type GenericCell struct {
-	chain    CellChain
-	level    CellLevel
-	priority CellPriority // priority of a cell is max of those of its children and itself
-	parent   Cell         // pointer to its parent cell
-	children CellList     // pointer to its children cells
+	chain              CellChain
+	level              CellLevel
+	priority           CellPriority
+	parent             Cell     // pointer to its parent cell
+	children           CellList // pointer to its children cells
+	atOrHigherThanNode bool     // true if the cell is at or higher than node level
+
+	totalGpuNum           int32                  // total GPU number of a cell
+	freeGpuNum            int32                  // free GPU number of a cell
+	freeGpuNumForPriority int32                  // free GPU number for a certain priority (lower priority considered as free)
+	usedGpuNumAtPriority  map[CellPriority]int32 // GPU number used by each priority
+	// GPU number used by the same priority (used for sorting cells in topologyAwareScheduler)
```

Comment:
>same [](start = 27, length = 4)

same with what? GenericCell.priority?

Reply:
Do not put temp var as fields 

---
In reply to: [332820470](https://github.com/microsoft/pai/pull/3684#discussion_r332820470) [](ancestors = 332820470)